Cauli/Florets/Inspector/Record List/InspectorTableViewController.swift

```diff
@@ -27,7 +27,9 @@ internal class InspectorTableViewController: UITableViewController {
     private static let recordPageSize = 20
 
     var cauli: Cauli
-    var records: [Record] = []
+    let dataSource = InspectorTableViewDatasource()
+
+    let searchController = UISearchController(searchResultsController: nil)
```

Comment:
What do you think about this pattern? This way we have the initialization and configuration in one place and independent of the initialization of the view. So the only thing left to be done is to actually use the UISearchController in the `viewDidLoad`.

```swift
private lazy var searchController: UISearchController = {
   let searchController = UISearchController(searchResultsController: nil)
   searchController.dimsBackgroundDuringPresentation = false
   searchController.hidesNavigationBarDuringPresentation = false
   searchController.searchBar.delegate = self
   searchController.searchBar.autocapitalizationType = .none
   searchController.searchBar.autocorrectionType = .no
   return searchController
}
```

Reply:
What is the reason for lazy here? If we call it in viewDidLoad it will be instantiated anyways, right?

Reply:
It if isn't lazy, it will be initialized during the initialization of the `InspectorTableViewController` instance. But, that will fail because we are using `self` from here on. So having it lazy means it will definitely be initialized (once we initialize the view).

Reply:
Fixed